release/downloadIstioCtl.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+#
+# Separate downloader for istioctl
+#
+# This file will be fetched as: curl -L https://git.io/downloadIstioCtl | sh -
+#
+detect_os() {
+  OS=$(uname)
+  case $OS in
+    'Linux')
+      OSEXT='linux'
+      FILENAME='istioctl'
+      ;;
+    'WindowsNT')
+      OSEXT='win.exe'
+      FILENAME='istioctl.exe'
+      ;;
+    'Darwin') 
+      OSEXT='osx'
+      FILENAME='istioctl'
+      ;;
+    *) ;;
+  esac
+}
+
+get_version() {
+  if [ "x${ISTIO_VERSION}" = "x" ] ; then
+  ISTIO_VERSION=$(curl -L -s https://api.github.com/repos/istio/istio/releases/latest | \
+                  grep tag_name | sed "s/ *\"tag_name\": *\"\\(.*\\)\",*/\\1/")
+  fi
+}
+
+download_istioctl() {
+  NAME="istioctl-${OSEXT}"
+  URL="https://storage.googleapis.com/istio-artifacts/${ISTIO_VERSION}/artifacts/istioctl/${NAME}"
+  echo "${URL}"
+  echo "Downloading ${NAME} ..."  
+  curl -L "${URL}" -o "${FILENAME}"
+  BINDIR="$(pwd)"
+  echo "Add $BINDIR to your path; e.g copy paste in your shell and/or ~/.profile:"
+  echo "export PATH=\"\$PATH:$BINDIR\""
```

Comment:
is this correct?  shouldn't it match the instruction here - https://istio.io/docs/setup/kubernetes/download/#download-and-prepare-for-the-installation?

Reply:
@linsun If I understand this correctly, the script I've created in this PR provides the functionality that matches the documentation, and will eventually map to `/getLatestIstioCtl`. The code in question here is the "inner curl command" that actually does the work of downloading the istioctl binary.

Reply:
@linsun basically, I'm following that same code pattern that allows the `/getLatestIstio`URL to work as expected.

Reply:
I would expect the instruction to be aligned with our doc, like "Add the istioctl client to your PATH environment variable, on a macOS or Linux system: $ export PATH=$PWD/bin:$PATH".

Could you paste the output of the cmd as part of your manual testing?

Also, Is it possible to add a test case for this?

Reply:
Oh, I was misunderstanding. You're referring to the echo statement on Line 40 of the script, not the curl cmd on Line 38.

Now that I'm clear, yes, I can re-word the echo statement to read more like what you suggested. Also, to answer your questions:

> Could you paste the output of the cmd as part of your manual testing?

Are you referring to the echo statement again? If so, yes, I can make sure that the echo statement reads like what we have in our docs, and I'll test that as part of my manual testing.

> Also, Is it possible to add a test case for this?

This would be a test case for ensuring that this script downloads the appropriate istioctl versions, correct? If so, I'll have to find the appropriate place to add the test, but yes, I think we can.

Reply:
@ibm-jason I was referring to the output when you run this download istioctl script.  I assume the cmd can run with your PR for a released istio like 1.1.2 for example?

Reply:
@linsun Yes, this script will run locally with a released version of Istio like 1.1.2, but with one caveat:

`ISTIO_VERSION` must be set to the _commit SHA_ that corresponds to the 1.1.2 release, since there not an istioctl artifact released with the `major.minor.patch` annotation. For example:

`https://storage.googleapis.com/istio-artifacts/1.1.2/artifacts/istioctl` does not exist currently, which is why I've  added a line in the Makefile to publish the istioctl binary with the RELEASE_TAG in addition to the git commit SHA..